src/main.ts

```diff
@@ -1,24 +1,20 @@
-import './style.css'
-import typescriptLogo from './typescript.svg'
-import viteLogo from '/vite.svg'
-import { setupCounter } from './counter.ts'
+/* eslint-disable no-console */
+import { createElement } from "./libs/jsx";
+import { Element } from "./Test";
 
-document.querySelector<HTMLDivElement>('#app')!.innerHTML = `
-  <div>
-    <a href="https://vite.dev" target="_blank">
-      <img src="${viteLogo}" class="logo" alt="Vite logo" />
-    </a>
-    <a href="https://www.typescriptlang.org/" target="_blank">
-      <img src="${typescriptLogo}" class="logo vanilla" alt="TypeScript logo" />
-    </a>
-    <h1>Vite + TypeScript</h1>
-    <div class="card">
-      <button id="counter" type="button"></button>
-    </div>
-    <p class="read-the-docs">
-      Click on the Vite and TypeScript logos to learn more
-    </p>
-  </div>
-`
+const A = () => createElement("div", null, "first");
 
-setupCounter(document.querySelector<HTMLButtonElement>('#counter')!)
+const B = Element();
+
+const Test = createElement(
+  "div",
+  { className: "test" },
+  createElement(A, null),
+  createElement("div", null, createElement("h1", null, "Hello Babel Test")),
+  createElement("p", null, "transformation"),
+);
+
+const element = Test;
+
+console.log(JSON.stringify(element, null, 2));
```

Comment:
[P2] 댓글에 설명드린대로 jsx 함수인 Element가 createElement를 통해서 virtual dom object로 변환되는 과정을 보는 것이라 console로 결과 보는 것은 아래 코드로 축소될 수 있을 것 같습니다!

```suggestion
const element = Element();
console.log(JSON.stringify(element, null, 2));
```

Reply:
const B의 결과를 확인하면서 다른 경우도 테스트해보다가 정리가 부족했습니다. 정리해서 올려두겠습니다.